### PR TITLE
Blaze: Add networking models for Blaze target options

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of: #11540 

## Description

This PR adds new models for Blaze target options and mappers for them.

Note: I'm using the same `ID` key for locations and their parent locations following the suggestion in peeHDf-2fw-p2#comment-1615. If for some reason the backend team doesn't like the idea, I'll update the model and mapper later.

## Testing instructions

Just CI passing is sufficient.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/BlazeTargetOptions.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public let description: String
+
+    public init(id: String, description: String) {
+        self.id = id
+        self.description = description
+    }
+}
+
+/// Location to target for a Blaze campaign.
+/// This has to be a class so that it can reference a property with the type of itself.
```
Comment (hafizrahman) [inline Networking/Networking/Model/BlazeTargetOptions.swift:53]: thanks for the comment 👍🏼

File: Networking/Networking/Model/BlazeTargetOptions.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    public let type: String
+
+    /// Post code of the location (if available)
+    public let code: String?
+
+    /// ISO code of the location (if available)
+    public let isoCode: String?
+
+    /// Parent of the location.
+    public let parentLocation: BlazeTargetLocation?
```
Comment (hafizrahman) [inline Networking/Networking/Model/BlazeTargetOptions.swift:73]: Suggestion: 

Add a bit more comment to clarify the meaning of "parent" in this context. Something like:
```
/// "Parent" refers to the larger geographical context of a given location. 
/// For example the parent of a city is a state, or the parent of a country is a continent.
```

**Review by hafizrahman — APPROVED** [review]
This looks good 👍🏼  I just added a suggestion.

File: Networking/Networking/Model/BlazeTargetOptions.swift (hunk 2 above)
Comment (itsmeichigo) [reply Networking/Networking/Model/BlazeTargetOptions.swift:73]: Thank you! I'll update this in #11546 to avoid triggering CI all over again 😅.

**Pull request merged**